### ISSUE TITLE
[super_context_menu] Open menu with a tap

### ISSUE THREAD

**Issue by nelsoncampos-cloudwalk** [superlistapp/super_native_extensions#347]
GM @knopp, I'd like to see the possibility to add the option of open the context menu trough a tap too, like the iMessage do when you send a phone number.

if possible, to indicate where I can change it in the code and open a PR to contribute with the package. I was reading but didn't understood if I need to change in flutter, or in rust.

Preview from iMessage
https://github.com/superlistapp/super_native_extensions/assets/60667230/76523efb-e006-49d2-996d-5831129a237a

Overall thank for this package, amazing delivery for the flutter community.

**Comment by AnAlpaca** [comment]
HI there, I am no expert here and I am not involved in the porject. But I did manage to get this to work by editting the following: 

In the _ContextMenuDetector class, which is found here: <Usr dir>/.pub-cache/hosted/pub.dev/super_context_menu-0.8.5/lib/src/desktop.dart

 ```
bool _acceptPrimaryButton() {
    final keys = RawKeyboard.instance.keysPressed;
    return defaultTargetPlatform == TargetPlatform.macOS &&
        keys.length == 1 &&
        keys.contains(LogicalKeyboardKey.controlLeft);
  }

  bool _canAcceptEvent(PointerDownEvent event) {
    if (event.kind != PointerDeviceKind.mouse) {
      return false;
    }
    if (event.buttons == kSecondaryButton ||
        event.buttons == kPrimaryButton && _acceptPrimaryButton()) {
      return widget.contextMenuIsAllowed(event.position);
    }

    return false;
  }
 ```
 
 From here you can edit till your hearts content, for me I just added another condition for the _acceptPrimaryButton method, where I checked if windows, and then created an _acceptSecondaryButton with the appropraite conditions to disable right click for that use case. Hope this helps. Obviulsy for any of this to work you would need to fork the repo or somehow get creative with extensions of the appropraite classes.

**Comment by nelsoncampos-cloudwalk** [comment]
Hey @AnAlpaca Thanks for your reply, that can be helpful for desktop features, I really appreciate.

I'm looking for a solution for mobile, more specific iOS implementation

**Comment by AnAlpaca** [comment]
Also just wanted to say that I really appreciate this package! Its amazing, and the desing is fantastic!

**Comment by knopp** [comment]
I don't think this is possible. On iOS the event is triggered from a system gesture recognizer. From the top of my head I'm not aware of any public API on iOS that can programatically show the context menu.

**Comment by nelsoncampos-cloudwalk** [comment]
Yes! I tried to do an example app in native code to see if it was possible and I didn't find any solution to it. just a menu button that has some differences from the context menu.

I created a plugin to show a UIButton with a menu on top of FlutterViewController as a subview, and did all the component lifecycle, but it being very tricky, to render and update the native element to be anchored in the flutter widget.

How would be your approach to implement the Menu Button @knopp? I saw an a open issue about that.

https://github.com/superlistapp/super_native_extensions/issues/189

**Comment by BeADre** [comment]
You can try [this](https://github.com/flutter/flutter/pull/137936#issuecomment-1955249270), It's useful for me.

**Comment by lukepighetti** [comment]
Would be great to be able to open this menu with a simple tap

**Comment by jtkeyva** [comment]
yes a tap would be useful. but it's probably by design. i think what we want is a menu, not necessarily a context menu. hmm

**Comment by daverin** [comment]
This would be so great

**Comment by uiYzzi** [comment]
I implemented it in the branch I forked, but I only tested it on macOS and iOS (iOS is not working).
https://github.com/uiYzzi/super_native_extensions
